packages/@aws-cdk/aws-sns/lib/topic-base.ts

```diff
@@ -102,4 +105,18 @@ export abstract class TopicBase extends Resource implements ITopic {
     });
   }
 
+  private getNextTokenId(scope: Construct) {
+    let nextToken = 1;
+    const re = /TokenSubscription:([\d]*)/gm;
+    // Search for previous subscriptions with unresolved tokens
+    for (const source of scope.node.findAll()) {
+      // Used regex to find the next available suffix
+      const m = re.exec(source.node.id);
+      if (m && +m[1] >= nextToken) {
```

Comment:
Curious: Is the prefix-ed `+` here converting the string to an integer? Is this a documented way of doing that conversion in typescript?

Reply:
It's a JavaScript convention: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators

"unary plus is the fastest and preferred way of converting something into a number, because it does not perform any other operations on the number."

Happy to change this if you have a more preferred way.

Reply:
@strazeadin Generally I think we try to lean towards more explicit. My vote here is for `Number` or if we are requiring int or float specifically, `parseInt` or `parseFloat` respectively. After some light reading there are some unique uses for unary +, but it doesn't seem needed in this case.

I learned something new though 👨‍🎓

Reply:
Appreciate the comment, I refactored it a bit to make it much easier to follow and more explicit